packages/expo-cli/src/commands/build-native/iOSBuilder.ts

```diff
@@ -62,7 +62,8 @@ class iOSBuilder implements Builder {
     const credentialsSource = await ensureCredentialsAsync(
       provider,
       this.buildProfile.workflow,
-      this.buildProfile.credentialsSource
+      this.buildProfile.credentialsSource,
+      nonInteractive
     );
     this.credentials = await provider.getCredentialsAsync(credentialsSource);
```

Comment:
you need to pass nonInteractive flag to the credentials provider (preferably in constructor)

With the current setup, you are not passing this flag to code responsible for generating credentials